graph/test_simple_graph.py

```diff
@@ -0,0 +1,248 @@
+
+
+def test_neighbors_not_found(test_graph):
+    with pytest.raises(KeyError) as e:
+        test_graph.neighbors(6)
+    assert 'Node not found' in str(e.value)
+
+
+def test_adjacent(test_graph):
+    assert test_graph.adjacent(42, "test")
+
+
+def test_adjacent_first_node_not_found(test_graph):
+    with pytest.raises(KeyError) as e:
+        test_graph.adjacent("test2", 42)
+    assert 'First node not found' in str(e.value)
+
+
+def test_adjacent_second_node_not_found(test_graph):
+    with pytest.raises(KeyError) as e:
+        test_graph.adjacent("test", 47)
+    assert 'Second node not found' in str(e.value)
+
+
+def test_depth_first_from_node_0(test_depth_traversal_graph):
+    assert test_depth_traversal_graph.depth_first_traversal(0) == [
+        0, 1, 4, 5, 6, 7, 8, 2, 9, 3]
+
+
+def test_depth_first_from_node_1(test_depth_traversal_graph):
+    assert test_depth_traversal_graph.depth_first_traversal(1) == [
+        1, 4, 5, 6, 7, 8]
+
+
+def test_depth_first_no_edges():
+    test = Graph()
+    test.add_node(55)
+    test.add_node("test")
+    test.add_node(2)
+    assert test.depth_first_traversal(55) == [55]
+
+
+def test_depth_multiple_edges():
+    test = Graph()
+    for i in range(5):
+        test.add_node(i)
+    test.add_edge(0, 1)
+    test.add_edge(0, 2)
+    test.add_edge(1, 2)
+    test.add_edge(1, 3)
+    test.add_edge(2, 3)
+    test.add_edge(3, 4)
+
+    assert test.depth_first_traversal(0) == [0, 1, 2, 3, 4]
+
+
+def test_depth_first_from_non_existent_node(test_depth_traversal_graph):
+    with pytest.raises(KeyError) as e:
+        test_depth_traversal_graph.depth_first_traversal(11)
+    assert "Node does not exist" in str(e.value)
+
+
+def test_depth_cyclic(test_graph):
+    assert test_graph.depth_first_traversal(5) == [5, 42, "test"]
+
+
+def test_breadth_first(test_breadth_traversal_graph):
+    assert (
+        test_breadth_traversal_graph.breadth_first_traversal(1) == range(1, 10))
+
+
+def test_breadth_first_from_node_3(test_breadth_traversal_graph):
+    assert test_breadth_traversal_graph.breadth_first_traversal(3) == [3]
+
+
+def test_breadth_first_from_non_existent_node(test_breadth_traversal_graph):
+    with pytest.raises(KeyError) as e:
+        test_breadth_traversal_graph.breadth_first_traversal(11)
+    assert "Node does not exist" in str(e.value)
+
+
+def test_breadth_cyclic(test_graph):
+    assert test_graph.breadth_first_traversal(5) == [5, 42, "test"]
+
+
+def test_breadth_first_no_edges():
+    test = Graph()
+    test.add_node(55)
+    test.add_node("test")
+    test.add_node(2)
+    assert test.breadth_first_traversal(55) == [55]
+
+
+def test_breadth_multiple_edges():
+    test = Graph()
+    for i in range(5):
+        test.add_node(i)
+    test.add_edge(0, 1)
+    test.add_edge(0, 2)
+    test.add_edge(1, 2)
+    test.add_edge(1, 3)
+    test.add_edge(2, 3)
+    test.add_edge(3, 4)
+
+    assert test.breadth_first_traversal(0) == [0, 1, 2, 3, 4]
```

Comment:
It'd be nice to see traversal tests for graphs with un-connected nodes, just to ensure that you don't have a way to mistakenly teleport between unconnected nodes.